### DISCUSSION TITLE
Help installing pydeck from source (WSL2)

### DISCUSSION THREAD

**Discussion by 6A61736F6E206E61646572** [visgl/deck.gl#6456] [Q&A] [answered]
Sorry in advance since this is probably something very trivial I am not familiar with (Javascript related)

[Docs](https://github.com/visgl/deck.gl/blob/master/bindings/pydeck/docs/installation.rst) say to:
```sh
git clone https://github.com/visgl/deck.gl
cd deck.gl/bindings/pydeck
yarn bootstrap
pip install .
```

However I am stuck at the `yarn` step:
```sh
> cd deck.gl/bindings/pydeck
> yarn bootstrap
00h00m00s 0/0: : ERROR: [Errno 2] No such file or directory: 'bootstrap'
```

Now I have little to no experience with yarn and other JS related tools but if yarn is parsing https://github.com/visgl/deck.gl/blob/master/package.json then bootstrap is one of the script entry points listed there so I am not sure why it is erroring like this. Is my version of yarn perhaps too old?

```sh
> yarn --version
0.32+git
```

**Comment by 6A61736F6E206E61646572** [comment, marked as answer]
Sorry for the noise, I managed to figure it out. In my WSL2 Ubuntu 21.04 container, `yarn` is actually pointing to a different tool that has nothing to do with Javascript, Running `yarnpkg bootstrap` worked.

Relevant upstream issue: https://github.com/yarnpkg/yarn/issues/673

Would it be better to rename the example commands in deck.gl docs to say yarnpkg instead of yarn? Or is this issue too niche?

**Reply by kylebarron** [reply]
I'm not sure. I don't have experience with WSL but on my machine the executable `yarn` always points to the JS package manager 🤷‍♂️

**Reply by 6A61736F6E206E61646572** [reply]
Looks like it's mainly Debian/Ubuntu where this occurs due to the distro maintainers not allowing the Javascript `yarn` takeover the name from the software that was using that name previously. On Arch Linux, `yarn` was the JS one.
